homework/index.js

```diff
@@ -0,0 +1,156 @@
+                    callback(new Error(`Network error: ${XHR.status} - ${XHR.statusText}`));
+                }
+
+            }
+        }
+
+    }
+
+
+    function renderReposList(error, reposObj) {
+
+        const header = document.querySelector('header');
+        if (error) {
+            const err = document.createElement('div');
+            header.parentNode.insertBefore(err, header.nextSibling);
+            err.innerText = error.message;
+            err.className = 'error';
+
+        } else {
+            reposObj.sort((a, b) => { return a.name.localeCompare(b.name); });
+            //create list and append it to header
+            const selectList = createAndAppend('select', header, 'id', 'selectList');
+            //default hidden option 'Select a Repository'
+            const selectOption = createAndAppend('option', selectList, 'selected', '', 'disabled', '');
+            selectOption.setAttribute('hidden', '');
+            selectOption.innerText = 'Select a Repository';
+
+            for (let rep in reposObj) {
+                // add an option for each repo with value that is his index at reposObject 
+                const selectOption = createAndAppend('option', selectList, 'value', rep);
+                selectOption.innerText = reposObj[rep].name;
+
+            }
+            //add a listener when select a new option
+            selectList.onchange = () => {
+                //get the value of the selected option
+                const value = document.getElementById('selectList').value;
+                renderInfo(reposObj[value]);// call renderInfo and pass the selected repo to show his info
+                //call fetchJSON to fetch repo contributors and pass renderContributions to show them
+                fetchJSON(reposObj[value].contributors_url, renderContributions);
+
+            }
+        }
+    }
+
+    function renderInfo(rep) {
+
+        // add repo data to the appropriate cells in info table.
+        const cell = document.querySelector('#info-section tr:nth-child(1) td:nth-child(2)');
+        cell.innerText = '';
+        const link = createAndAppend('a', cell, 'href', rep.html_url, 'target', '_blank');
+        link.innerText = rep.name;
+        document.querySelector('#info-section tr:nth-child(2) td:nth-child(2)').innerText = rep.description;
+        document.querySelector('#info-section tr:nth-child(3) td:nth-child(2)').innerText = rep.forks;
+        document.querySelector('#info-section tr:nth-child(4) td:nth-child(2)').innerText = rep.updated_at.substring(0, 10);
+
+    }
+
+    function renderContributions(error, contributions) {
+
+        if (error) {
+
+            const err = document.createElement('div');
+            document.querySelector('header').parentNode.insertBefore(err, header.nextSibling);
+            err.innerText = error.message;
+            err.className = 'error';
+
+        } else {
+
+            const ele = document.getElementById('contributions-container');
+
```

Comment:
Try to give variable names so that they describe the value properly, 'ele' is not a very descriptive name.